src/Sentry.Unity.Editor.iOS/SentryXcodeProject.cs

```diff
@@ -59,41 +85,68 @@ internal void ReadFromProjectFile()
                 throw new FileNotFoundException("Could not locate generated Xcode project at", _projectPath);
             }
 
-            _project.ReadFromString(File.ReadAllText(_projectPath));
-            _mainTargetGuid = _project.GetUnityMainTargetGuid();
-            _unityFrameworkTargetGuid = _project.GetUnityFrameworkTargetGuid();
+            _project.GetType().GetMethod("ReadFromString", BindingFlags.Public | BindingFlags.Instance).Invoke(_project, new[] { File.ReadAllText(_projectPath) });
+            _mainTargetGuid = (string)_project.GetType().GetMethod("GetUnityMainTargetGuid", BindingFlags.Public | BindingFlags.Instance).Invoke(_project, null);
+            _unityFrameworkTargetGuid = (string)_project.GetType().GetMethod("GetUnityFrameworkTargetGuid", BindingFlags.Public | BindingFlags.Instance).Invoke(_project, null);
         }
 
         public void AddSentryFramework()
         {
             var relativeFrameworkPath = Path.Combine("Frameworks", FrameworkName);
-            var frameworkGuid = _project.AddFile(relativeFrameworkPath, relativeFrameworkPath);
-
-            _project.AddFrameworkToProject(_mainTargetGuid, FrameworkName, false);
-            _project.AddFrameworkToProject(_unityFrameworkTargetGuid, FrameworkName, false);
-
-            _project.AddFileToEmbedFrameworks(_mainTargetGuid, frameworkGuid); // Embedding the framework because it's dynamic and needed at runtime
+            var frameworkGuid = (string)_project.GetType()
+                .GetMethod("AddFile", BindingFlags.Public | BindingFlags.Instance)
+                .Invoke(_project, new object[] { relativeFrameworkPath, relativeFrameworkPath, 1 }); // 1 is PBXSourceTree.Source
+
+            var addFrameworkToProjectMethod = _project.GetType()
+                .GetMethod("AddFrameworkToProject", BindingFlags.Public | BindingFlags.Instance);
+            addFrameworkToProjectMethod.Invoke(_project, new object[] { _mainTargetGuid, FrameworkName, false });
+            addFrameworkToProjectMethod.Invoke(_project, new object[] { _unityFrameworkTargetGuid, FrameworkName, false });
+
+            // Embedding the framework because it's dynamic and needed at runtime
+            // _project.AddFileToEmbedFrameworks(_mainTargetGuid, frameworkGuid);
+            var xcodeAssembly = Assembly.Load("UnityEditor.iOS.Extensions.Xcode");
```

Comment:
The only place where we need the extension of the Xcode extension. Should that loading code move into the constructor?